vowpalwabbit/vwdll.cpp

```diff
@@ -59,23 +59,21 @@ VW_DLL_PUBLIC VW_HANDLE VW_CALLING_CONV VW_InitializeEscaped(const char16_t* pst
 
 
 VW_DLL_PUBLIC VW_HANDLE VW_CALLING_CONV VW_InitializeA(const char * pstrArgs)
-{ string s(pstrArgs);
-  VW::workspace* all = VW::initialize(s);
+{
+  VW::workspace* all = VW::initialize(static_cast<std::string>(pstrArgs));
```

Comment:
I really don't think that cast is safe. What about constructing a local string was an issue before?

Reply:
The local string wasn't being freed an unrecognized option exception was thrown. Wrapping the logic here in a `try...catch` also works to solve the issue

Reply:
It doesn't really make sense why the string would not get cleaned up. The exception would just cause the stack to unwind right? What else could be going on here?